src/libraries/System.Threading.RateLimiting/tests/SlidingWindowRateLimiterTests.cs

```diff
@@ -0,0 +1,713 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Threading.RateLimiting.Test
+{
+    public class SlidingWindowRateLimiterTests : BaseRateLimiterTests
+    {
+        [Fact]
+        public override void CanAcquireResource()
+        {
+            var limiter = new SlidingWindowRateLimiter(new SlidingWindowRateLimiterOptions(1, QueueProcessingOrder.NewestFirst, 1,
+                TimeSpan.Zero, 2, autoReplenishment: false));
+            var lease = limiter.Acquire();
+
+            Assert.True(lease.IsAcquired);
+            Assert.False(limiter.Acquire().IsAcquired);
+
+            lease.Dispose();
+            Assert.False(limiter.Acquire().IsAcquired);
+            Assert.True(limiter.TryReplenish());
+            Assert.True(limiter.TryReplenish());
+
+            Assert.True(limiter.Acquire().IsAcquired);
+        }
+
+        [Fact]
+        public override void InvalidOptionsThrows()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>(
+                () => new SlidingWindowRateLimiterOptions(-1, QueueProcessingOrder.NewestFirst, 1, TimeSpan.FromMinutes(2), 1, autoReplenishment: false));
+            Assert.Throws<ArgumentOutOfRangeException>(
+                () => new SlidingWindowRateLimiterOptions(1, QueueProcessingOrder.NewestFirst, -1, TimeSpan.FromMinutes(2), 1, autoReplenishment: false));
+            Assert.Throws<ArgumentOutOfRangeException>(
+                () => new SlidingWindowRateLimiterOptions(1, QueueProcessingOrder.NewestFirst, 1, TimeSpan.FromMinutes(2), -1, autoReplenishment: false));
+        }
+
+        [Fact]
+        public override async Task CanAcquireResourceAsync()
+        {
+            var limiter = new SlidingWindowRateLimiter(new SlidingWindowRateLimiterOptions(2, QueueProcessingOrder.NewestFirst, 4,
+                TimeSpan.Zero, 2, autoReplenishment: false));
+
+            using var lease = await limiter.WaitAsync();
+
+            Assert.True(lease.IsAcquired);
+            var wait = limiter.WaitAsync(2);
+            Assert.False(wait.IsCompleted);
+
+            Assert.True(limiter.TryReplenish());
+
+            Assert.False(wait.IsCompleted);
+
+            var wait2 = limiter.WaitAsync(2);
+            Assert.False(wait2.IsCompleted);
+
+            Assert.True(limiter.TryReplenish());
+
+            Assert.True((await wait2).IsAcquired);
+        }
+
+        [Fact]
+        public async Task CanAcquireMulitpleRequestsAsync()
```

Comment:
Can we add comments in this test? I already forgot what scenario it was testing specifically